### PR TITLE
Next Release v0.4.2

### PR DESCRIPTION
**Pull request by srivarra**
- small dataset loading fix
- bumped version from 0.4.1 -> 0.4.2

**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Creates a the next release `0.4.2`.

**How did you implement your changes**

Adjusted instances of `0.4.1` to `0.4.2`. Added a tiny fix for downloading datasets.

**Remaining issues**

Next step is to create a new release via the releases page.

### CODE REVIEW

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Some minor tweaks. 

Will you update requirements.txt to have a comment at the top that says a new release needs to be made anytime the requirements are changed?

Will you also add a comment in the developer docs explaining that code changes aren't automatically propagated into the docker, with the steps that need to be done and the code chunk they'll use. I believe @ackagel mentioned it in his docker update PR

File: ark/utils/data_utils.py [hunk 1]
```diff
@@ -550,7 +550,7 @@ def download_example_data(save_dir: Union[str, pathlib.Path]):
     The following is a link to the dataset used:
     https://huggingface.co/datasets/angelolab/ark_example
 
-    The dataset will be saved in `{save_dir}/example_dataset/image_data`.
```
[the remark this answers is not in the document]
Comment (srivarra) [reply ark/utils/data_utils.py:553]: I just got that mixed up, that'll be reverted.

File: ark/utils/data_utils.py [hunk 2]
```diff
@@ -559,7 +559,7 @@ def download_example_data(save_dir: Union[str, pathlib.Path]):
     # Downloads the dataset
     ds = datasets.load_dataset("angelolab/ark_example")
 
-    data_path = pathlib.Path(ds["base_dataset"]["Data Path"][0]) / "input_data"
```
Comment (ngreenwald) [inline ark/utils/data_utils.py:562]: This is the actual path that huggingface produces. Just changing it here will cause an error if it's not changed in the raw files, right?

File: docs/_rtd/contributing.md [hunk 3]
```diff
@@ -40,6 +40,6 @@ This means that some of the tests didn't pass. You can click on the link for mor
...
+No one writes perfect code the first time. Chances are, your reviewer will have some suggested changes for your code. Take the time to carefully read through their comments, and make sure to ask any clarifying questions. Then, once you understand what's being asked of you, update your PR with the requested changes. You can continue to make commits to the existing branch that you used to create your PR. As you push new commits to that branch, the PR will automatically update, and Travis will automatically re-run to test this new code. Once you've addressed the comments made during review, you can click the 're-request review' button on the top. Once your reviewer is satisfied with your changes, they will approve your PR and merge it into the main branch.
```
Comment (ngreenwald) [inline docs/_rtd/contributing.md:43]: Did you do a control-f for all instances of master to make sure you go them all?

**Review by ngreenwald — CHANGES_REQUESTED** [review]

File: docs/_rtd/contributing.md (hunk 3 above)
Comment (srivarra) [reply docs/_rtd/contributing.md:43]: Yes

File: ark/utils/data_utils.py (hunk 2 above)
Comment (srivarra) [reply ark/utils/data_utils.py:562]: It's changed in the hugging face repo as well [here](https://huggingface.co/datasets/angelolab/ark_example/blob/main/ark_example.py). It was inconsistent with what we have here in the notebooks.

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Alex just made a similar change in his PR that got merged in

**Comment by srivarra** [comment]
Yeah I let Alex know which files to checkout from this branch since the changes to the huggingface dataset file would cause a couple of tests to fail without the adjustment on his end.

**Comment by ngreenwald** [comment]
Looks good, gonna wait to see if @ackagel's -u update flag fix will require a change to the docker so that we can merge that in before putting out a new release today

**Comment by ngreenwald** [comment]
Turns out the issue with the docker -u flag was that the docker image was built using the old docker file. Will you add a comment to the docker file saying that any changes will require a new release? Then we'll be good to merge this in

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Almost there!

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**